### DISCUSSION TITLE
v1.0.0 Release Notes 🎉

### DISCUSSION THREAD

**Discussion by endigo9740** [skeletonlabs/skeleton#1125] [Announcements]
![v1-release](https://user-images.githubusercontent.com/1509726/224212919-20d0291f-8444-4655-9833-27cf5ef3fd9b.jpg)

# Introduction

Please join us in welcoming Skeleton v1.0! Today's release is a huge milestone for the project. It has been a year since the project was started, and in that time we've had 50+ contributors, closed 500 GitHub Issues, 500 pull requests, and provided 27 pre-release versions. These are not meaningless numbers, but rather represent the cumulative effort of contributors to deliver what we feel is the best UI library available for Svelte today.

## Launch Day Events

To kick things off we'll be celebrating with not one but TWO livestream events today. View the details here:

-   https://github.com/skeletonlabs/skeleton/discussions/1087

## What should you expect from v1.0?

The v1.0 release eases the project into a point of stability. In other words, you can now have the confidence to build and use Skeleton in your production projects. No software is perfect though, which means we will continue releasing regular maintenance updates and introducing additional features and extended capabilities over time. Major changes can still occur, but we'll do our best to limit these to major point releases (ex: v2.0, v3.0, etc)

## Special Thanks

A special thanks to [Trey Weir](https://github.com/weirbones) and [Thomas Jespersen](https://github.com/thomasbjespersen) for helping get Skeleton off the ground. To our core team: @niktek, @Sarenor, @ryceg, @luke-hagar-sp, @AdrianGonz97, @JustBarnt, @SebasF1349 for helping make the project better every day. And of course our wonderful community, contributors, and sponsors that have provided unwavering support along the way. Seriously, thank you all so much!

# Release Notes

Release: https://github.com/skeletonlabs/skeleton/releases/tag/1.0.0

Summary:

-   🎉 v1.0.0 released!
-   📖 Massive documentation overhaul
-   🖼️ Many new previews and examples
-   🐞 A plethora of improvements and bugfixes

## Documentation Overhaul - https://github.com/skeletonlabs/skeleton/issues/783

The primary focus of today's release is a massive overhaul of the Skeleton documentation. This was accomplished through an incredible grassroots movement within our community to help improve every page, top to bottom. Special thanks for everyone that reviewed and submitted changes during this time.

A very big thank you to our biggest contributors: @Sarenor, @niktek, @RyanTheGuy, @DukeRupert and @oatnog

### Homepage Redesign - https://github.com/skeletonlabs/skeleton/issues/1070

![home](https://user-images.githubusercontent.com/1509726/224210740-12be0203-04dd-47b7-9eb6-7d7de8ea2795.png)

The Skeleton documentation homepage has been rebuilt from the ground up. In addition to a brand new visual design, it now serves as an introduction and showcase of the many key features that make Skeleton unique. Of course the entire page is built with Skeleton's adaptive theming system, which means it looks great regardless of what theme you choose!

The page also includes a link to launch Skeleton on StackBlitz, allowing you to easily test drive the entire library without installation. If you're a fan of what you create in this sandbox, link the project to GitHub and continue using it all the way to production.

### Page Structure

![doc-structure](https://user-images.githubusercontent.com/1509726/224210760-ed86a436-281a-4868-a73d-2626a0359ce4.png)

The structure of each inner documentation page has been reworked to provide more information at a glance. Quickly find the name, description, and import statement of each feature. As well as link to the page source, feature source, accessibility guidelines, and more. Then quickly navigate the page with the included table of contents. Hover over a title section to generate a deep link straight to that specific topic of interest.

### Improved Previews - https://github.com/skeletonlabs/skeleton/issues/1031

![Screenshot 2023-03-09 at 8 52 47 PM](https://user-images.githubusercontent.com/1509726/224211270-dc06002e-18b7-4c6b-80ef-294f00a8d901.png)

The Skeleton documentation now also includes many more previews and demonstrations of each feature. With an interactive preview component to test on different color backgrounds and direct access to quick configurable settings. Code snippets are also co-located with each preview, allowing you to quickly replicate each example in your project.

### Information Quality

![Screenshot 2023-03-09 at 8 53 27 PM](https://user-images.githubusercontent.com/1509726/224211345-97c50a0d-27f0-44cf-8b44-d7ad0fb47ba8.png)

We've put a tremendous amount of effort to better explain what Skeleton is, walk you through getting started, and of course cover the most essential features of the library, such as props, design tokens, and variants. Navigation has also been streamlined to make it easier to traverse from section to section.

### Improved Search

![Screenshot 2023-03-09 at 8 54 23 PM](https://user-images.githubusercontent.com/1509726/224211487-469f1cf9-88d6-4939-af4a-cacceda2666b.png)

Documentation search has also received a "glow up", allowing you to quickly trigger search, query, and navigate to a feature page using full keyboard navigation. It's never been easier to find what you're looking for.

### Additional Documentation Updates

-   Content audits by the community:
    https://github.com/skeletonlabs/skeleton/pull/1051
    https://github.com/skeletonlabs/skeleton/pull/1054
    https://github.com/skeletonlabs/skeleton/pull/1055
    https://github.com/skeletonlabs/skeleton/pull/1056
    https://github.com/skeletonlabs/skeleton/pull/1059
    https://github.com/skeletonlabs/skeleton/pull/1071
    https://github.com/skeletonlabs/skeleton/pull/1084
    https://github.com/skeletonlabs/skeleton/pull/1086
    https://github.com/skeletonlabs/skeleton/pull/1089
-   Various typo and grammar fixes:
    https://github.com/skeletonlabs/skeleton/issues/1008
    https://github.com/skeletonlabs/skeleton/issues/1015
    https://github.com/skeletonlabs/skeleton/pull/1035
    https://github.com/skeletonlabs/skeleton/pull/1040
    https://github.com/skeletonlabs/skeleton/pull/1073
    https://github.com/skeletonlabs/skeleton/pull/1085
    https://github.com/skeletonlabs/skeleton/pull/1121
-   Enabled unstyled exceptions for ins/del typography:
    https://github.com/skeletonlabs/skeleton/issues/1060
-   Fixed a broken link within the footer:
    https://github.com/skeletonlabs/skeleton/issues/1010
-   The table of Contents now update selected page tab:
    https://github.com/skeletonlabs/skeleton/issues/1052
-   Documented a number of missing component properties:
    https://github.com/skeletonlabs/skeleton/pull/1108
-   Audited and fixed a number of responsive design issues:
    https://github.com/skeletonlabs/skeleton/issues/1058

## Polish and Improvements

-   Popups should now be much more reliable and performant:
    https://github.com/skeletonlabs/skeleton/issues/1018
-   Improved IntelliSense info for the AppShell header slot:
    https://github.com/skeletonlabs/skeleton/pull/1083
-   A vertical radio group example added:
    https://github.com/skeletonlabs/skeleton/pull/1091
-   Improved the click handler for App Rail Tile:
    https://github.com/skeletonlabs/skeleton/pull/1110
-   Added the missing class attribute for the Radio Item:
    https://github.com/skeletonlabs/skeleton/issues/1119

### Bugfixes

-   Bugfix for default Lightswitch height property:
    https://github.com/skeletonlabs/skeleton/issues/1003
-   Bugfix for Input Chip button type:
    https://github.com/skeletonlabs/skeleton/pull/1063
-   Bugfix for input group Select > Option background color on Windows:
    https://github.com/skeletonlabs/skeleton/issues/1041
-   Bugfix for File Dropzone binding files:
    https://github.com/skeletonlabs/skeleton/issues/1021
-   Bugfix for File Dropzone selectable area in Firefox:
    https://github.com/skeletonlabs/skeleton/pull/1109
-   Bugfix for button group hover state extending too far in Safari:
    https://github.com/skeletonlabs/skeleton/issues/1096
-   Bugfix to theme generator hex color input:
    https://github.com/skeletonlabs/skeleton/issues/1111

**Comment by zakaria-chahboun** [comment]
Congratulations :tada:

**Reply by EssaAlshammri** [reply]
:)
